content-src/components/Bookmarks/Bookmarks.js

```diff
@@ -0,0 +1,95 @@
+const React = require("react");
+const {connect} = require("react-redux");
+const {justDispatch} = require("common/selectors/selectors");
+const {actions} = require("common/action-manager");
+const classNames = require("classnames");
+const {FormattedMessage} = require("react-intl");
+const {SpotlightItem} = require("components/Spotlight/Spotlight");
+const getBestImage = require("common/getBestImage");
+
+const PlaceholderBookmarks = React.createClass({
+  render() {
+    return (<div className="bookmarks-placeholder">
+      <FormattedMessage id="header_bookmarks_placeholder" />
+    </div>);
+  }
+});
+
+const Bookmarks = React.createClass({
+  getDefaultProps() {
+    return {
+      length: 3,
+      page: "NEW_TAB",
+      placeholder: false
+    };
+  },
+  getInitialState() {
+    return {isAnimating: false};
+  },
+  onClickFactory(index, site) {
+    return () => {
+      let payload = {
+        event: "CLICK",
+        page: this.props.page,
+        source: "BOOKMARKS",
+        action_position: index,
+        highlight_type: site.type,
+        metadata_source: site.metadata_source
+      };
+      this.props.dispatch(actions.NotifyEvent(payload));
+    };
+  },
+  // XXX factor out into a stateless component
+  renderSiteList() {
+    const sites = this.props.sites.filter(site => site.bookmarkGuid)
+                                  .slice(0, this.props.length);
+
+    return sites.map((site, i) =>
+      <SpotlightItem
+        index={i}
+        key={site.guid || site.cache_key || i}
+        page={this.props.page}
+        source="BOOKMARKS"
+        bestImage={getBestImage(site.images)}
+        onClick={this.onClickFactory(i, site)}
+        dispatch={this.props.dispatch}
+        {...site}
+        prefs={this.props.prefs} />
+    );
+  },
+  handleHeaderClick() {
```

Comment:
Yeah, so this collapsing stuff can be removed if you use the new `<CollapsibleSection/>` instead.